r2dbc-spec/src/main/asciidoc/data-types.adoc

```diff
@@ -126,21 +135,28 @@ Driver implementations should implement the following type mapping and can suppo
 | Represents a time without a date part and without timezone.
 | `java.time.LocalTime`
 
+| `TIME WITH TIME ZONE`
+| Represents a time with a timezone offset.
+| `java.time.OffsetTime`
+
 | `TIMESTAMP`
 | Represents a date/time without a timezone.
 | `java.time.LocalDateTime`
 
-| `TIMESTAMP` with Timezone Offset
+| `TIMESTAMP WITH TIME ZONE`
 | Represents a date/time with a timezone offset.
 | `java.time.OffsetDateTime`
 
-| `TIMESTAMP` with Timezone
-| Represents a date/time with a timezone.
-| `java.time.ZonedDateTime`
+|===
+
+[[datatypes.mapping.intervals]]
+.SQL Type Mapping for Interval Types
+|===
+|SQL Type|Description |Java Type
 
-| `INTERVAL`
-| Interval date types such as `YEAR`, `MONTH`, `DAY`, `HOUR` and similar representing a time quantity.
-| `java.time.Duration`
+| `INTERVAL(p)`
```

Comment:
`(p)` part is not correct. `INTERVAL` data types have a qualifier and only the qualifier may have optional precision. Valid qualifiers are `YEAR`, `MONTH`, `DAY TO HOUR`, etc.

There are 2 major incompatible types of intervals with subtypes.

1. Year-month intervals:
`INTERVAL YEAR`
`INTERVAL MONTH`
`INTERVAL YEAR TO MONTH`
2. Day-time intervals:
`INTERVAL DAY`
`INTERVAL HOUR`
`INTERVAL MINUTE`
`INTERVAL SECOND`
`INTERVAL DAY TO HOUR`
`INTERVAL DAY TO MINUTE`
`INTERVAL DAY TO SECOND`
`INTERVAL HOUR TO MINUTE`
`INTERVAL HOUR TO SECOND`
`INTERVAL MINUTE TO SECOND`

They can also have optional leading field precision and/or fractional seconds precision like `INTERVAL DAY(4) TO SECOND(3)` or `INTERVAL SECOND(5, 3)`.

I think you need to list year-month and day-time intervals separately.

However, PostgreSQL also has own plain `INTERVAL` data type that can have all 6 datetime fields. There is no standard mapping for it in Java.

Reply:
Looking at all the variations of `INTERVAL` I'm wondering whether we should touch on specifying. Probably we should wait until we have to come up with a standard mapping.

Reply:
You don't need to list all of them. There are year-month intervals that can be represented using `Period` and daytime intervals that can be represented using `Duration`. Additional details aren't necessary here.

On the other hand, JDBC does not provide mappings for these data types.